packages/firestore/src/exp/reference_impl.ts

```diff
@@ -258,20 +260,20 @@ export function setDoc<T>(
  */
 export function setDoc<T>(
   reference: DocumentReference<T>,
-  data: Partial<T>,
+  data: NestedPartial<T>,
   options: SetOptions
 ): Promise<void>;
 export function setDoc<T>(
   reference: DocumentReference<T>,
-  data: T,
+  data: WithFieldValue<T> | NestedPartial<T>,
```

Comment:
NesterPartial is a superset of WithFieldValue, isn't it? I don't think it is required here.